File: 백준/Silver/28278. 스택 2/스택 2.py
import sys
input = sys.stdin.readline

n = int(input())
stack = []
for i in range(n):
    command = input().rstrip()

    if len(command) > 2:
        stack.append(int(command[2:]))
    elif command == '2':
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif command == '3':
        print(len(stack))
    elif command == '4':
        print(1 if len(stack) == 0 else 0)
    elif command =='5':
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: 백준/Silver/28278. 스택 2/스택 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int num = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack();
        StringBuilder sb = new StringBuilder();
        for(int i = 0 ; i<num ; i++){
            String command = br.readLine();
            st = new StringTokenizer(command);

            switch (st.nextToken()){
                case "1" :
                    stack.push(Integer.parseInt(st.nextToken()));
                    break;
                case "2":
                    if(stack.isEmpty()){
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(stack.pop()).append("\n");
                    }
                    break;
                case "3" :
                    sb.append(stack.size()).append("\n");
                    break;
                case "4" :
                    if(stack.isEmpty()) {
                        sb.append(1).append("\n");
                    } else{
                        sb.append(0).append("\n");
                    }
                    break;
                case "5" :
                    if(stack.isEmpty()){
                        sb.append(-1).append("\n");
                    } else{
                        sb.append(stack.peek()).append("\n");
                    }
                    break;
            }
        }
        System.out.println(sb.toString());
    }
}
